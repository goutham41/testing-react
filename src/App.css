.App {
  text-align: center;
}
.btn{
  padding:10px;
  border-radius:5px;
  color:whitesmoke;
  margin-top: 20px;
}
.borderded{
  border:1px solid gray;
}
.ghost{
  border:none;
}
.red{
  background-color:rgb(189, 82, 82);
}
.green{
  background-color: rgb(91, 192, 192);
}