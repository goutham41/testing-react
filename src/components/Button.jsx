import React from 'react'
import "../App.css";
const Button = ({ children, colorScheme, variant, onClick }) => {
  return (
    <button
      onClick={onClick}
      data-testid="cbtn"
      className={`btn ${colorScheme || "red"} ${variant || "ghost"}`}
    >
      {children}
    </button>
  );
};

export default Button