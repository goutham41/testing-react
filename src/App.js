import React,{useState} from "react";
import "./App.css";
import Button from "./components/Button";
import Counter from "./Day3testingReact/Counter";

function App() {
  const [theme, setTheme] = useState("light");
  return (
    <div className="App">
      <h3>Theme is {theme}</h3>
      <Button
        onClick={() => {
          setTheme(theme === "light" ? "dark" : "light");
        }}
        colorScheme="green"
      >
        change Theme
      </Button>
      <br />
      <Counter/>
    </div>
  );
}

export default App;
