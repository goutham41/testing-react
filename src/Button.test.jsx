import { render, screen, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import Button from "./components/Button";
import App from "./App";
//intgration system
describe("ButtonTests", function () {
  it("should have button in dom", function () {
    render(<Button>Click me</Button>);
    let button = screen.getByText("Click me");
    expect(button).toBeInTheDocument();
    //    screen.debug()
  });

  it("should render based on testid", function () {
    render(<Button>Click me</Button>);
    let button = screen.getByTestId("cbtn");
    expect(button).toBeInTheDocument();
  });

  it("should render empty button", function () {
    render(<Button></Button>);
    let button = screen.getByTestId("cbtn");
    expect(button).toBeEmptyDOMElement();
  });

  it("should change theme", function () {
    render(<App></App>);
    let h3 = screen.getByText("Theme is light");
    expect(h3).toBeInTheDocument();
    expect(h3).toHaveTextContent("light");

    let themeButton = screen.getByText("change Theme");
    //  userEvent.click(themeButton);userEvent or you can fireEvent
    fireEvent.click(themeButton);
    expect(h3).toHaveTextContent("dark");
  });

  it("should call given function", function () {
    const mockfn = jest.fn();
    render(<Button onClick={mockfn}>lick me</Button>);
    const btn = screen.getByTestId("cbtn");
    fireEvent.click(btn);
    fireEvent.click(btn);
    expect(mockfn).toBeCalledTimes(2);
  });

  //Snapshot
  it("Should match snapshot", function () {
    const tree = renderer
      .create(
        <Button colorScheme={"green"} variant="ghost">
          ADD counter
        </Button>,
      )
      .toTree();
    expect(tree).toMatchSnapshot();
  });
});
