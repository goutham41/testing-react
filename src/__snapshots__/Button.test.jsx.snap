// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonTests Should match snapshot 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "children": "ADD counter",
    "colorScheme": "green",
    "variant": "ghost",
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "children": "ADD counter",
      "className": "btn green ghost",
      "data-testid": "cbtn",
      "onClick": undefined,
    },
    "rendered": Array [
      "ADD counter",
    ],
    "type": "button",
  },
  "type": [Function],
}
`;
